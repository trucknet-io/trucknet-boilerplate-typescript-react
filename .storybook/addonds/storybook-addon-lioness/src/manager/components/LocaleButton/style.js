export const defaultStyle = {
    height: '50px',
    width: '100px',
    padding: '5px',
    border: 0,
    borderRight: '1px solid #d3d3d3',
    borderBottom: '1px solid #d3d3d3',
    background: 'none',
    lineHeight: '30px',
    textAlign: 'center',
    textTransform: 'uppercase',
    transitionProperty: 'background',
    transitionDuration: '100ms',
    transitionTimingFunction: 'linear'
};

export const activeStyle = {
    background: '#f7f7f7',
    fontWeight: 'bold'
};

export default function (props) {
    return {
        ...defaultStyle,
        ...(props.active ? activeStyle : {})
    };
};
