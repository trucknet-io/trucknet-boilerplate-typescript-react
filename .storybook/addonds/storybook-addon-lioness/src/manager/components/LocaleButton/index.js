import React from 'react';
import PropTypes from 'prop-types';
import style from './style';

class LocaleButton extends React.Component {
    constructor () {
        super();

        this.handleClick = this.handleClick.bind(this);
    }

    handleClick (event) {
        event.preventDefault();

        this.props.onClick(this.props.locale);
    }

    render () {
        return (
            <button
                type="button"
                style={style(this.props)}
                onClick={this.handleClick}
            >
                {this.props.locale}
            </button>
        );
    };
}

LocaleButton.propTypes = {
    locale: PropTypes.string.isRequired,
    active: PropTypes.bool,
    onClick: PropTypes.func.isRequired
};

export default LocaleButton;
