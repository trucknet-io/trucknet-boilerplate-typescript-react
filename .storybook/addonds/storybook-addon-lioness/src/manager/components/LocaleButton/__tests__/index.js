import React from 'react';
import renderer from 'react-test-renderer';

import LocaleButton from '../';

describe('LocaleButton', () => {
    test('renders correctly', () => {
        const locale = 'en';
        const onClick = () => {};

        const tree = renderer.create(
            <LocaleButton locale={locale} onClick={onClick} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('handles click correctly', () => {
        const locale = 'en';
        const onClick = jest.fn();
        const event = { preventDefault: jest.fn() };

        const component = renderer.create(
            <LocaleButton locale={locale} onClick={onClick} />
        );

        const tree = component.toJSON();
        tree.props.onClick(event);

        expect(event.preventDefault.mock.calls).toHaveLength(1);
        expect(onClick.mock.calls).toHaveLength(1);
        expect(onClick.mock.calls[0][0]).toBe(locale);
    });
})
