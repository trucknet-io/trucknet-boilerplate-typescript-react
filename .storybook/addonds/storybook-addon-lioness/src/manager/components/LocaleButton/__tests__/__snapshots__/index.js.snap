// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleButton renders correctly 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "background": "none",
      "border": 0,
      "borderBottom": "1px solid #d3d3d3",
      "borderRight": "1px solid #d3d3d3",
      "height": "50px",
      "lineHeight": "30px",
      "padding": "5px",
      "textAlign": "center",
      "textTransform": "uppercase",
      "transitionDuration": "100ms",
      "transitionProperty": "background",
      "transitionTimingFunction": "linear",
      "width": "100px",
    }
  }
  type="button"
>
  en
</button>
`;
